fendermint/vm/topdown/src/sync/syncer.rs

```diff
@@ -248,9 +249,12 @@ where
             tracing::debug!(height, "adding data to the cache");
 
             self.provider.new_parent_view(height, Some(data.clone()))?;
-            self.vote_tally
-                .add_block(height, Some(data.0.clone()))
-                .map_err(map_voting_err)?;
+            if let Some(p) = self.provider.sealed_proposal_at_height(height)? {
```

Comment:
Out of interest what does the seal_proposal_at_height do?

Reply:
It calculates the commitment of the proposal at the target height, kind of sealed it which means no more updates allowed.